src/library/scala/collection/Seq.scala

```diff
@@ -266,15 +266,17 @@ trait SeqOps[+A, +CC[_], +C] extends Any
     *  @return `true` if this $coll has `that` as a suffix, `false` otherwise.
     */
   def endsWith[B >: A](that: Iterable[B]): Boolean = {
-    val i = iterator.drop(length - that.size)
-    val j = that.iterator
-    while (i.hasNext && j.hasNext)
-      if (i.next() != j.next())
-        return false
+    if (this.sizeCompare(that) < 0) false else {
+      val i = iterator.drop(length - that.size)
+      val j = that.iterator
 
-    !j.hasNext
-  }
+      while (i.hasNext && j.hasNext)
+        if (i.next() != j.next())
+          return false
 
+      !j.hasNext
```

Comment:
Maybe just personal preference, but wouldn't it be more readable to add:
```
!j.hasNext && !i.hasNext
```
At the bottom to be sure iterator ends with that iterator, instead of adding the if logic on top that you did now?

Reply:
Nvm, christmas braindead, that doesn't solve the bug.

Reply:
are you sure? it seems like it would solve it to me